src/H5Tcompound.c

```diff
@@ -426,77 +426,81 @@ H5Tpack(hid_t type_id)
 herr_t
 H5T__insert(H5T_t *parent, const char *name, size_t offset, const H5T_t *member)
 {
-    unsigned idx; /* Index of member to insert */
-    size_t   total_size;
-    unsigned i;                   /* Local index variable */
-    herr_t   ret_value = SUCCEED; /* Return value */
+    unsigned            idx; /* Index of member to insert */
+    size_t              total_size;
+    unsigned            i;                   /* Local index variable */
+    herr_t              ret_value = SUCCEED; /* Return value */
+    H5T_shared_t *const msh = member->shared, *const psh = parent->shared;
```

Comment:
I like the readability of the use of temp vars, however personally I'd prefer declarations on individual lines.